sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -592,6 +594,48 @@ def __init__(self, special_parameter):
                                   check_estimator, MyEstimator)
 
 
+class LRXFailTagButCheckPasses(LogisticRegression):
+    def _more_tags(self):
+        return {"_xfail_checks": {
+            # check_complex_data actually passes for this estimator
+            "check_complex_data": "This is a bad check"}
+        }
+
+
+def test_check_estimator_xfail_tag_warns():
+    # When a _xfail_checks check passes, warns stating that the test
+    # passes and can be removed from the tag
+
+    estimators = [LRXFailTagButCheckPasses(), LRXFailTagButCheckPasses]
+
+    for estimator in estimators:
+        assert_warns_message(FutureWarning,
+                             "LRXFailTagButCheckPasses:check_complex_data "
+                             "passed, it can be removed from the "
+                             "_xfail_check tag",
+                             check_estimator, estimator)
+
+
+class LRXFailDoesNotRaiseOnComplex(LRXFailTagButCheckPasses):
+    def fit(self, X, y):
+        # do not raise error for complex check
+        try:
+            return super().fit(X, y)
+        except ValueError as e:
+            if "Complex data not supported" not in str(e):
+                raise
+        return self
+
+
+def test_check_estimator_xfail_tag_skips():
```

Comment:
My understanding is that check_estimator does not skip anything?

Reply:
This checks that it raises a "SkipTestWarning", so `test_check_estimator_xfail_tag_raises_skip_test_warning" ?